sdk/dotnet/Pulumi.Automation/DictionaryContentsComparer.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2019, Pulumi Corporation
+
+using System.Collections.Generic;
+
+namespace Pulumi.Automation
+{
+    /// Compares two dictionaries for equality by content, as F# maps would.
+    internal sealed class DictionaryContentsComparer<K, V> : IEqualityComparer<IDictionary<K, V>> where K : notnull
```

Comment:
Style nitpick - I don't think this should be in the root directory of the project if it isn't public / exported